Deprecate `secrets` field in `type: custom` authentication in `Kafka` CR (#146)
strimzi/proposals: 092-deprecate-secrets-field-in-custom-server-authentication.md

# Deprecate `secrets` field in `type: custom` authentication in `Kafka` CR

This proposal suggests to deprecate and later remove the `secrets` field of the `type: custom` authentication in the Kafka custom resource.

## Motivation

When using [`type: custom` authentication](https://strimzi.io/docs/operators/latest/full/configuring.html#type-KafkaListenerAuthenticationCustom-reference) (introduced in Strimzi 0.28) in Kafka brokers, users can mount any Secret resources that are used by their custom authentication mechanism.
These Secrets are mounted into the `/opt/kafka/custom-authn-secrets/custom-listener-<listener_name>-<port>/<secret_name>` directory and users can reference them by file path in their configuration.
Strimzi does not do any validation or sanitization of their content and does not have any special handling for these Secrets.
It just mounts them into the container.

Recently, we introduced support for mounting custom volumes to any Strimzi container.
This feature was introduced by [SP075 - Support for additional volumes](https://github.com/strimzi/proposals/blob/main/075-additional-volumes-support.md) in Strimzi 0.43.

> It allows to mount Secrets, but also Config Maps, PVCs or CSI volumes.

Reviewer: ```suggestion
It allows the mounting of Secrets, but also Config Maps, PVCs or CSI volumes.
```

Similarly to the Secrets from the `type: custom` authentication, Strimzi does not do any validation or sanitization of their content and does not have any special handling for these Secrets.
They are just mounted into the `/mnt` path where they can be used by the user.

The additional volumes feature can be used to replace the `secrets` field in the `type: custom` authentication.

> Having two different features to cover the same thing seems to be unnecessary

Reviewer: ```suggestion
Having two different features to cover the same thing is unnecessary
```
